File: frontend/src/types/flashcards.ts
export interface FlashcardDeck {
  id: string;
  class_id: string;
  session_id: string | null;
  title: string;
  public: boolean;
  created_by: string;
  created_at: string;
}

export interface Flashcard {
  id: string;
  deck_id: string;
  question: string;
  answer: string;
  topic: string | null;
  created_by: string;
  created_at: string;
  updated_at: string;
}

File: backend/app/schemas/flashcards.py
from pydantic import BaseModel
from typing import Optional

class CreateDeckRequest(BaseModel):
    class_id: str
    session_id: Optional[str] = None
    title: str
    public: bool = True

class GenerateFlashcardsRequest(BaseModel):
    note_id: str
    count: int = 20

class CreateCardRequest(BaseModel):
    question: str
    answer: str
    topic: Optional[str] = None

class UpdateCardRequest(BaseModel):
    question: Optional[str] = None
    answer: Optional[str] = None
    topic: Optional[str] = None

